docs/intro/tutorial.rst

```diff
@@ -1,141 +1,446 @@
+to determine which page object class to use based on its input parameters (URL
+and item class).
+
+Your web-poet framework can take care of everything else:
+
+#.  It matches the input URL and item class to ``BookPage``, based on the URL
+    pattern that you defined with the :meth:`~web_poet.overrides.handle_urls`
+    decorator, and the return type that you declared in the page object class
+    (``Book``).
+
+#.  It inspects the inputs declared by ``BookPage``, and builds an instance of
+    ``BookPage`` with the required inputs.
+
+    ``BookPage`` is a :class:`~web_poet.pages.WebPage` subclass, and
+    :class:`~web_poet.pages.WebPage` declares an attribute named ``response``
+    of type :class:`~web_poet.page_inputs.HttpResponse`. Your web-poet
+    framework sees this, and creates an
+    :class:`~web_poet.page_inputs.HttpResponse` object from the input URL as a
+    result, by downloading the URL response, and assigns that object to the
+    ``response`` attribute of a new ``BookPage`` object.
+
+#.  It builds the output item, ``Book(title='The Exiled')``, using the
+    :meth:`~web_poet.pages.ItemPage.to_item` method of ``BookPage``, inherited
+    from :class:`~web_poet.pages.ItemPage`, which in turn uses all fields of
+    ``BookPage`` to create an instance of ``Book``, which you declared as the
+    return type of ``BookPage``.
+
+.. _scrapy-poet: https://scrapy-poet.readthedocs.io
+
+
+Extend and override your code
+=============================
+
+To continue this tutorial, you will need extended versions of ``Book`` and
+``BookPage``, with additional fields. However, rather than editing the existing
+``Book`` and ``BookPage`` classes, you will see how you can instead create new
+classes that inherit them.
+
+Append the following code to ``tutorial-project/tutorial/items.py``:
+
+.. literalinclude:: /tutorial-project/tutorial/items.py
+   :language: python
+   :lines: 9-15
+
+The code above defines a new item class, ``CategorizedBook``, that inherits the
+``title`` attribute from ``Book`` and defines 2 more attributes: ``category``
+and ``category_rank``.
+
+Append the following code to
+``tutorial-project/tutorial/pages/books_toscrape_com.py``:
+
+.. literalinclude:: /tutorial-project/tutorial/pages/books_toscrape_com.py
+   :language: python
+   :lines: 16, 19-23, 25, 29-32
+
+In the code above:
+
+-   You define a new page object class: ``CategorizedBookPage``.
 
+-   ``CategorizedBookPage`` subclasses ``BookPage``, inheriting its ``title``
+    field, and defining a new one: ``category``.
 
-    response = requests.get("http://books.toscrape.com")
+    ``CategorizedBookPage`` does *not* define a ``category_rank`` field
+    yet, you will add it later on. For now, the default value defined in
+    ``CategorizedBook`` for ``category_rank``, ``None``, will be used.
 
-Creating Page Input
-===================
+-   ``CategorizedBookPage`` indicates that it returns a ``CaterogizedBook``
```

Comment:
```suggestion
-   ``CategorizedBookPage`` indicates that it returns a ``CategorizedBook``
```

Reply:
My first attempt at committing this change locally:

```$ git comimt -am "CaterogizedBook → CategorizedBook"```

:facepalm: